### PR TITLE
Feature 14162 ask for dni

### PR DESCRIPTION
**Pull request by pepeportela**

### CODE REVIEW

**Comment by mreyk** [comment]
@pepeportela The commit d1bbda8 shouldn't been done. You have should rebase `feature-14159-DNI-field` branch instead.

Undo the commit (with `git commit revert d1bbda8`) and rebase the `feature-14159-DNI-field` branch with:

```
git checkout feature-14159-DNI-field
git pull
git checkout feature-14162-ask-for-DNI
git rebase feature-14159-DNI-field
```

**Comment by pepeportela** [comment]
Solved conflicts

**Comment by mreyk** [comment]
@pepeportela If you call it `national_id`, keep calling it `national_id`. Don't use the term `dni`, it only makes sense in our country. Use the term `national_id` to make it more generic.

File: lms/djangoapps/courseware/views/views.py [hunk 1]
```diff
@@ -739,9 +740,14 @@ def _progress(request, course_key, student_id):
     grade_summary = course_grade.summary
 
     studio_url = get_studio_url(course, 'settings/grading')
-
+
```
Comment (mreyk) [inline lms/djangoapps/courseware/views/views.py]: Remove tab.

**Comment by pepeportela** [comment]
Removed the dni name

File: lms/djangoapps/courseware/views/views.py [hunk 2]
```diff
@@ -743,16 +744,23 @@ def _progress(request, course_key, student_id):
...
         'courseware_summary': courseware_summary,
         'studio_url': studio_url,
         'grade_summary': grade_summary,
         'staff_access': staff_access,
         'student': student,
+        'national_id': national_id,
         'passed': is_course_passed(course, grade_summary),
         'credit_course_requirements': _credit_course_requirements(course_key, student),
-        'certificate_data': _get_cert_data(student, course, course_key, is_active, enrollment_mode)
+        'certificate_data': _get_cert_data(student, course, course_key, is_active, enrollment_mode, national_id)
```
Comment (mreyk) [inline lms/djangoapps/courseware/views/views.py]: @pepeportela Do not break API function call. You can get national_id inside the `_get_cert_data` function with the `student` variable (User).

File: lms/djangoapps/courseware/views/views.py [hunk 3]
```diff
@@ -761,7 +769,7 @@ def _progress(request, course_key, student_id):
     return response
 
 
-def _get_cert_data(student, course, course_key, is_active, enrollment_mode):
+def _get_cert_data(student, course, course_key, is_active, enrollment_mode, national_id):
```
Comment (mreyk) [inline lms/djangoapps/courseware/views/views.py]: @pepeportela Do not break API function call. You can get national_id inside the `_get_cert_data` function with the `student` variable (User).

**Comment by pepeportela** [comment]
Removed national_id in function call

File: lms/djangoapps/extrainfo/models.py [hunk 4]
```diff
@@ -1,13 +1,20 @@
+import logging
```
Comment (mreyk) [inline lms/djangoapps/extrainfo/models.py]: @pepeportela This library is not being used. Remove it.

File: lms/djangoapps/extrainfo/models.py [hunk 5]
```diff
@@ -1,13 +1,20 @@
...
+import json
```
Comment (mreyk) [inline lms/djangoapps/extrainfo/models.py]: @pepeportela This library is not being used. Remove it.

File: lms/djangoapps/extrainfo/models.py [hunk 6]
```diff
@@ -1,13 +1,20 @@
...
 from django.conf import settings
 from django.db import models
+from django.http import (
```
Comment (mreyk) [inline lms/djangoapps/extrainfo/models.py]: @pepeportela This library is not being used. Remove it.

File: lms/djangoapps/extrainfo/models.py [hunk 7]
```diff
@@ -1,13 +1,20 @@
...
+        HttpResponse, HttpResponseBadRequest, HttpResponseForbidden, HttpRequest
```
Comment (mreyk) [inline lms/djangoapps/extrainfo/models.py]: @pepeportela These libraries are not being used. Remove them.

File: lms/djangoapps/extrainfo/models.py [hunk 8]
```diff
@@ -1,13 +1,20 @@
...
+)
+from student.models import UserProfile
+
+logger = logging.getLogger(__name__)
```
Comment (mreyk) [inline lms/djangoapps/extrainfo/models.py]: @pepeportela This `logger` is not being used. Remove it.

File: lms/djangoapps/extrainfo/models.py [hunk 9]
```diff
@@ -1,13 +1,20 @@
+import logging
+import json
 from django.conf import settings
 from django.db import models
+from django.http import (
+        HttpResponse, HttpResponseBadRequest, HttpResponseForbidden, HttpRequest
+)
+from student.models import UserProfile
```
Comment (mreyk) [inline lms/djangoapps/extrainfo/models.py]: @pepeportela This library is not being used. Remove it.

File: lms/djangoapps/extrainfo/views.py [hunk 10]
```diff
@@ -0,0 +1,33 @@
+import logging
+from django.contrib.auth.models import User
+from django.http import (
+            HttpResponse, HttpResponseBadRequest, HttpResponseForbidden, HttpRequest
```
Comment (mreyk) [inline lms/djangoapps/extrainfo/views.py]: @pepeportela `HttpRequest` is not being used. Remove it.

**Comment by pepeportela** [comment]
Removed the libraries

**Pull request merged**